bolt-modules/prompt/lib/puppet/functions/prompt.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+# Display a prompt and wait for a response.
+#
+# > **Note:** Not available in apply block
+Puppet::Functions.create_function(:prompt) do
+  # Wait until targets are available.
+  # @param prompt The prompt to display.
+  # @param options A hash of additional options.
+  # @option options [Boolean] sensitive Disable echo back and mark the response as sensitive.
+  # @return The response to the prompt.
+  # @example Prompt the user if plan execution should continue
+  #   $response = prompt('Continue executing plan? [Y\N]')
+  # @example Prompt the user for sensitive information
+  #   $password = prompt('Enter your password', 'sensitive' => true)
+  dispatch :prompt do
+    optional_param 'String[1]', :prompt
+    optional_param 'Hash[String[1], Any]', :options
+    return_type 'Variant[String, Sensitive]'
+  end
+
+  def prompt(prompt = '', options = {})
+    unless Puppet[:tasks]
+      raise Puppet::ParseErrorWithIssue
+        .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING,
+                              action: 'prompt')
+    end
+
+    unless STDIN.tty?
+      issue = Puppet::Pops::Issues.issue(:NO_TTY) { 'STDIN is not a tty, unable to prompt' }
+      raise Puppet::ParseErrorWithIssue.from_issue_and_stack(issue, action: 'prompt')
+    end
+
+    executor = Puppet.lookup(:bolt_executor)
+    executor.report_function_call(self.class.name)
+
+    response = executor.prompt(prompt, options)
```

Comment:
Is the purpose of delegating to the `executor` here so that the function can fail in PE without having to modify the Puppet side?

If that's the case, then we should move the above `tty?` check into the executor method, otherwise that will always be the error you get in PE. I guess then it shouldn't be a `Puppet` error.